pkg/policy/l4.go

```diff
@@ -479,21 +497,25 @@ func (l4 *L4Filter) GetListener() string {
 	return l4.Listener
 }
 
-// ToMapState converts filter into a MapState with two possible values:
-//
-//	Entry with ProxyPort = 0: No proxy redirection is needed for this key
-//	Entry with any other port #: Proxy redirection is required for this key,
-//	                             caller must replace the ProxyPort with the actual
-//	                             listening port number.
+// toMapState converts a single filter into a MapState entries added to 'mapState'.  'filter' is
```

Comment:
Added a function type for this:
```
// 'mapStateEntryCallback' is a function called for each entry before adding to a MapState. If the
// function returns 'true', the entry is added, otherwise not. The function gets a reference to the
// entry so that it may update it's value (e.g., the proxy port).
type mapStateEntryCallback func(key Key, value *MapStateEntry) bool
```